### ISSUE TITLE
GitHubDesktop 64-bit installs git.exe that is only compatible with 32 bit cygwin1.dll

### ISSUE THREAD

**Issue by pksublime** [desktop/desktop#2075]
### Description

I recently installed GitHubDesktop 64bit and was getting a cygheap mismatch error when trying to run commands using the git.exe that was included. I double checked that my 64 bit cygwin was up to date and it was.
As a last resort I installed 32 bit cygwin and reset my path to direct to it and the git.exe started working.

### Version

**GitHub Desktop version:** 0.6.1

**OS version:** Microsoft Windows [Version 6.1.7601]

### Steps to Reproduce

1. Install 64 bit cygwin
2. Install 64 bit GitHubDesktop Beta
3. set path to include c:\cygwin64\bin
4. open a command prompt
5. navigate to a git repo
6. execute c:\Users\<user>\AppData\Local\GitHubDesktop\app-0.6.1\resources\app\git\cmd\git.exe fetch
7. observer error

```
   2145 [main] cat (15040) C:\Users\<user>\AppData\Local\GitHubDesktop\app-0.6.1\resources\app\git\usr\bin\cat.exe: *** fatal error - cygheap base mismatch detected - 0x180304408/0x1802FF408.
This problem is probably due to using incompatible versions of the cygwin DLL.
Search for cygwin1.dll using the Windows Start->Find/Search facility
and delete all but the most recent version.  The most recent version *should*
reside in x:\cygwin\bin, where 'x' is the drive on which you have
installed the cygwin distribution.  Rebooting is also suggested if you
are unable to find another cygwin DLL.
bash: line 1: 14156 Segmentation fault      cat > /dev/tty
error: failed to execute prompt script (exit code 139)
```
8. install 32 bit cygwin
9. redirect path to c:\cygwin\bin
10. repeat step 6 and observe resolution

**Expected behaviour:** I expected the 64 bit GitHubDesktop to work with 64 bit cygwin

**Actual behaviour:** 64 bit GitHubDesktop requires 32 bit cygwin

**Reproduces how often:** 100%

**Comment by shiftkey** [comment]
> I recently installed GitHubDesktop 64bit and was getting a cygheap mismatch error when trying to run commands using the git.exe that was included.

A few things to note before I dig deeper into this:

 - This isn't a supported use case of GitHub Desktop - the bundled Git is for use in the app only, with No Guarantees™ for other scenarios.
 - I also haven't tested Desktop alongside with cygwin (64-bit or 32-bit). If Git fails in those cases with the error, then we need to fix it.
 - If there's no error with Desktop, then I'd like to know more about your use case, because the source of it is [this project](https://github.com/desktop/dugite-native) and it's a build of Git that's designed for embedding in apps - it's got a lot of things removed that people might expect from Git.

**Comment by pksublime** [comment]
Due to my inability to connect to my GitHub Enterprise server through the gui (#2076), I was investigating what errors the git provided with the GUI was producing.

Currently my organization has us install Git via installing GitHub Desktop. Before this beta, that also installed "Git Shell". This allows our developers to use either a command line interface or a GUI to interact with our GitHub Enterprise server. If the user then also wanted to use other GUIs (most commonly Atlassian SourceTree or TortoiseGit), we directed those tools to also use the git.exe that was bundled with GitHub Desktop. That way the user will only have one executable to worry about with only one set of configurations.

>I also haven't tested Desktop alongside with cygwin (64-bit or 32-bit). If Git fails in those cases with the error, then we need to fix it.

This confuses me since the executable clearly needs cygwin1.dll and there isn't one within the c:\Users\<user>\AppData\Local\GitHubDesktop folder.

**Comment by shiftkey** [comment]
> Due to my inability to connect to my GitHub Enterprise server through the gui (#2076), I was investigating what errors the git provided with the GUI was producing.

The error you were seeing in #2076 is part of the API lookup we do to check the server exists, that you can sign in, and then authorize Desktop Electron. There might be subsequent SSL issues with Git operations, but they should share the same root cause.

>  If the user then also wanted to use other GUIs (most commonly Atlassian SourceTree or TortoiseGit), we directed those tools to also use the git.exe that was bundled with GitHub Desktop. That way the user will only have one executable to worry about with only one set of configurations.

Some additional context:

Because the install path for Desktop Classic would change between updates, we weren't able to properly support integrating with other tools that needed a `git.exe`. And when it came time for us to evaluate the command line experience for Desktop Electron, we had a long discussion in #340 and ultimately decided to recommend people install Git for Windows for their command line setup - the steps listed here show how to get the equivalent settings we had for "Git Shell": https://github.com/desktop/desktop/issues/340#issuecomment-302021683.

There were a couple of reasons for this:

 - installer size - we wanted to keep things as lean as possible, and there were a number of utilities in Git for Windows that we didn't need for Desktop - hence the slimmed down package
 - platform differences - macOS has a Git shim which sets things up on Windows, and it's recommended to install Git via your package manager on Linux. And while we had a lot of love from users about Git Shell (I still love it), Git for Windows has matured a lot since we first saw the need for Git Shell, so that's what I'd recommend using for the moment.

> This confuses me since the executable clearly needs cygwin1.dll and there isn't one within the `c:\Users<user>\AppData\Local\GitHubDesktop` folder.

There's a rich backstory on how Git for Windows interoperates with MSYS (which is based on Cygwin - the summary [here](https://github.com/git-for-windows/git/wiki/The-difference-between-MINGW-and-MSYS2#git-for-windows-2x-mingw-and-msys2) is a good starting point for understanding the relationship) but Git for Windows (and the Git bundled with Desktop) is intended to be self-contained, and having Cygwin on the PATH has been responsible for some interoperability issues in the past.

**Comment by shiftkey** [comment]
With `C:\cygwin64\bin` on my PATH i'm still able to clone and do regular Git operations in the Desktop app, so I believe everything here is working as expected in terms of Desktop working alongside an existing Cygwin installation:

<img width="541" src="https://user-images.githubusercontent.com/359239/27510151-e90333b6-58bf-11e7-97cf-0a7f64582fd3.png">

As direct usage of `git.exe` here isn't supported, we're using #304 to talk about the Git Shell story.

I've also reopened #2076 to confirm it's fixed by the `0.6.2` release that we published a couple of days ago.
